src/test/scala/org/broadinstitute/dsde/workbench/sam/util/DistributedLockSpec.scala

```diff
@@ -46,6 +46,20 @@ class DistributedLockSpec extends AsyncFlatSpec with Matchers with TestSupport {
     res.unsafeToFuture()
   }
 
+  it should "fail on withLock if there's same lock has already been set within 30 seconds" in {
+    assume(databaseEnabled, databaseEnabledClue)
+
+    val lockDetails = genLock.sample.get
+    val res = lockResource.use { lock =>
+      for {
+        _ <- lock.withLock(lockDetails).use(_ => IO.pure(()))
+        _ <- IO.sleep(2 seconds)
+        failed <- lock.withLock(lockDetails).use(_ => IO.pure(())).attempt
+      } yield failed.swap.toOption.get.asInstanceOf[FailToObtainLock].getMessage shouldBe s"can't get lock: $lockDetails"
+    }
+    res.unsafeToFuture()
+  }
+
```

Comment:
this unit test fails; I wrote the test to understand what the lock is actually doing